docs/en/cpp/guide/mavlink_direct.md

```diff
@@ -0,0 +1,226 @@
+
+## Getting Started
+
+### Basic obstacle example
+
+Let's assume we are a companion computer that detects obstacles and therefore needs to send the message [OBSTACLE_DISTANCE](https://mavlink.io/en/messages/common.html#OBSTACLE_DISTANCE) to the autopilot.
+
+```cpp
+// Set up as companion computer
+auto config = Mavsdk::Configuration{ComponentType::CompanionComputer};
+
+// Connect ...
+
+// Create MavlinkDirect plugin instance
+auto mavlink_direct = MavlinkDirect{system.value()};
+
+// Create a HEARTBEAT message
+MavlinkDirect::MavlinkMessage obstacle_distance{};
+obstacle_distance.message_name = "OBSTACLE_DISTANCE";
+obstacle_distance.system_id = config.get_system_id(); // Your component's system ID
+obstacle_distance.component_id = config.get_component_id(); // Your component's component ID
+obstacle_distance.target_system_id = 0; // Does not apply for this message
+obstacle_distance.target_component_id = 0; // Does not apply for this message
+
+// For now we just hard-code values for demonstration purposes.
+// We assume the sensor has is looking forward and has 8 segments with 10 degrees per segment,
+// so 80 degrees field of view.
+// On 3 segments we see an obstacle, the closest is at 7m.
+//
+// In a real application this JSON message content would of course be
+// assembled manually, or using a json library like jsoncpp or nlohmann/json.
+obstacle_distance.fields_json = R"({
+    "time_usec": 12345678,
+    "sensor_type": 3,
+    "distances": [
+        2000,   2000,  2000,  1000,   800,   700,  2000,  2000,
+        65536, 65536, 65536, 65536, 65536, 65536, 65536, 65536,
+        65536, 65536, 65536, 65536, 65536, 65536, 65536, 65536,
+        65536, 65536, 65536, 65536, 65536, 65536, 65536, 65536,
+        65536, 65536, 65536, 65536, 65536, 65536, 65536, 65536,
+        65536, 65536, 65536, 65536, 65536, 65536, 65536, 65536,
+        65536, 65536, 65536, 65536, 65536, 65536, 65536, 65536,
+        65536, 65536, 65536, 65536, 65536, 65536, 65536, 65536,
+        65536, 65536, 65536, 65536, 65536, 65536, 65536, 65536
+    ],
+    "increment": 0,
+    "min_distance": 000,
+    "max_distance": 2000,
+    "increment_f": 10.0,
+    "angle_offset": -40.0,
+    "frame": 12
+})";
+
+while (true) {
+    // Send the message
+    auto result = mavlink_direct.send_message(obstacle_distance);
+    if (result == MavlinkDirect::Result::Success) {
+        std::cout << "OBSTACLE_DISTANCE message sent successfully" << std::endl;
+    } else {
+        std::cerr << "OBSTACLE_DISTANCE message could not be sent: " << result << std::endl;
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
+}
+```
+
+Check the full example on [GitHub](https://github.com/mavlink/MAVSDK/tree/main/examples/mavlink_direct_sender).
+
+### Receiving Messages
+
+#### Subscribe to Specific Message Types
+
```

Comment:
How can it subscribe to a message that isn't defined, or is it defined? It might be that some messages are always available?